internal/v1alpha2/store/interface.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"archive/tar"
+	"errors"
+
+	"github.com/spf13/afero"
+)
+
+// The Store interface defines the required methods that must be implemented
+// by the bundle deployment store. These methods facilitate the unpacking and storage
+// of contents from all the sources.
+type Store interface {
+	afero.Fs
```

Comment:
I'm surprised you want to inline all the low-level bits from this.

Reply:
Right now, the sources use limited number of helpers just for creating/removing/verifying contents inside dir. Instead of specifying them separately, inlining of `afero.Fs` would make it easier and save us from adding more helpers to this interface. I assume we would be using this only in rukpak, that too to access localdir, so expecting an afero.Fs implementation to be passed seemed reasonable.